### PR TITLE
DOC: Adds documentation on building docs

### LINKED ISSUE #564: Contributing docs doesn't include method to build docs
I followed the guidelines at https://ml.dask.org/contributing.html:

```
$ conda env create -f ci/environment-3.7.yaml --name=dask-ml-dev
$ python -m pip install -e ".[dev]"
```

This worked. I made some documentation changes, so I had to build the docs. Typing `make html` in the `docs/` directory yielded some errors, and I had to install extra packages:

```
$ cd docs
$ make html
$  pip install IPython
$ make html
$ pip install jupyter_client
```

### PR DESCRIPTION
**Pull request by stsievert**
**What does this PR do?**
Closes #564 and adds documentation on how to build docs.

I ran into an issue running `conda env create -f ci/environment-docs.yaml --name=dask-ml-dev-docs`, some issue with xgboost and running a `brew install` command. That issue isn't relevant here, but I sidestepped it with the mentioned method (and I believe it also performs avoids some computation).

### CODE REVIEW

File: docs/source/contributing.rst [hunk 1]
```diff
@@ -138,6 +138,27 @@ Documentation
 
 We use `numpydoc <http://numpydoc.readthedocs.io/en/latest/format.html>`_ for our docstrings.
 
+Building the docs is possible with
+
+.. code-block:: none
+
+   $ conda env create -f ci/environment-docs.yaml --name=dask-ml-dev-docs
+
+This will build the full documentation after a complete installation. For a
```
Comment (TomAugspurger) [inline docs/source/contributing.rst]: I'd prefer to just remove the remainder of this. I'm in the process of migrating all our examples to examples.dask.org, and removing ones that aren't relevant.

Comment (stsievert) [reply docs/source/contributing.rst]: Removed.

**Comment by stsievert** [comment]
I'd appreciate some flag to not run the examples. They take a long time, and I don't need to run the examples every time I build the docs.

**Comment by TomAugspurger** [comment]
They're being removed in #369, so that won't be an issue anymore :)

**Comment by TomAugspurger** [comment]
Thanks!

**Pull request merged**